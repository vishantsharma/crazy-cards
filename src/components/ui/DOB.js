import React from 'react';
import Input from './Input';

const DateOfBirth = ({data, type, handleChange}) => {
    return (
        <>
        <Input 
          name="day"
          type={type}
          placeholder="DD"
          value={data.dob.day}
          onChange={handleChange} />
        
          <Input 
          name="month"
          type={type}
          placeholder="MM"
          value={data.dob.month}
          onChange={handleChange} />

          <Input 
          name="year"
          type={type}
          placeholder="YYYY"
          value={data.dob.year}
          onChange={handleChange} />
        </>
    )
}

export default DateOfBirth;