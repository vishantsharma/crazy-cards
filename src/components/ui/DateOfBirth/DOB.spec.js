import React from 'react';
import TestRenderer from 'react-test-renderer';
import DOB from './DOB';

describe('DOB', () => {
    let props = {
        data: {
            dob: {
                day: '',
                month: '',
                year: ''
            }
        },
        type: '',
        errors: '',
        className: '',
        handleChange: jest.fn().mockReturnValue()
    }
    it('should render', () => {
        const component = TestRenderer.create(<DOB {...props} />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})