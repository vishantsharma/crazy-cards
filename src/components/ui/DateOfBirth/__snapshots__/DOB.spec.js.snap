// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOB should render 1`] = `
Array [
  <label
    htmlFor="day"
  />,
  <input
    className=""
    id="day"
    name="day"
    onChange={[MockFunction]}
    placeholder="DD"
    type=""
    value=""
  />,
  <label
    htmlFor="month"
  />,
  <input
    className=""
    id="month"
    name="month"
    onChange={[MockFunction]}
    placeholder="MM"
    type=""
    value=""
  />,
  <label
    htmlFor="year"
  />,
  <input
    className=""
    id="year"
    name="year"
    onChange={[MockFunction]}
    placeholder="YYYY"
    type=""
    value=""
  />,
]
`;
